### ISSUE TITLE
"angularfire2": ^4.0.0-rc.0 -- Unhandled Promise rejection: No provider for AngularFireAuth

### ISSUE THREAD

**Issue by ravivit9** [angular/angularfire2#978]
### Version info

**Angular:**: 4.1.2

**Firebase:**: 3.9.0

**AngularFire:** : 4.0.0-rc.0

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**
    "@angular/animations": "^4.1.2",
    "@angular/common": "^4.1.2",
    "@angular/compiler": "^4.1.2",
    "@angular/core": "^4.1.2",
    "@angular/forms": "^4.1.2",
    "@angular/http": "^4.1.2",
    "@angular/material": "^2.0.0-beta.3",
    "@angular/platform-browser": "^4.1.2",
    "@angular/platform-browser-dynamic": "^4.1.2",
    "@angular/platform-server": "^4.1.2",
    "@angular/router": "^4.1.2",
    "angularfire2": "^4.0.0-rc.0",
    "core-js": "^2.4.1",
    "firebase": "^3.9.0",
    "geofire": "^4.1.2",
    "hammerjs": "^2.0.8",
    "ng2-flex-layout": "^0.1.5",
    "ng2-validation": "^4.2.0",
    "rxjs": "^5.4.0",
    "typescript": "^2.3.2",
    "zone.js": "^0.8.10"

### How to reproduce these conditions
Encountered error while restarting the app.  Code compiles successfully but the app throws runtime error as seen the screenshot.
 
**Failing test unit, Plunkr, or JSFiddle demonstrating the problem**

**Steps to set up and reproduce**

**Sample data and security rules**

<-- include/attach/link to some json sample data (or provide credentials to a sanitized, test Firebase project) -->

### Debug output

** Errors in the JavaScript console **

** Output from `firebase.database().enableLogging(true);` **

** Screenshots **
![screen shot 2017-05-15 at 16 21 26](https://cloud.githubusercontent.com/assets/3435829/26065023/959f515e-398a-11e7-87ed-90f30705bfb5.png)

### Expected behavior
App should show the login page.  The app worked perfectly all right prior to angularfire2 upgrade.  Unfortunately I am not sure where to point exactly as the error message doesn't gives a clue to point you to the right direction.

### Actual behavior
Black page with runtime error in console.

**Comment by markgoho** [comment]
Hi @ravivit9 -- general errors like this should definitely be researched first on StackOverflow. Getting a "No provider" error is associated with a missing import in your module file. Check to make sure that you've properly imported the `AngularFireAuthModule` and added it to the list of imports.

**Comment by ravivit9** [comment]
Thanks Mark.

I was looking for the solution past 3 days but couldn't find the right solution.  Normally I am mindful creating an issue in this forum as this platform shouldn't be polluted with lots of general question but as a last, I did created the issue but quick after that I found the solution as below: CLOSING the issue.

I got the solution from the following AF2 issue:
https://github.com/angular/angularfire2/issues/956
